### PR TITLE
Add new fn `Sftp::from_session` & integration test with `openssh`

### PR DESCRIPTION
**Pull request by NobodyXu**
- Add new feature `openssh-sftp-client/openssh`
 - Rename feature `openssh-sftp-client/ci-tests` to `__ci-tests`
   so that docs.rs would ignore this feature.
 - Simulate docs.rs doc building process in `check.sh`
 - Add optional dep `openssh` v0.9.9 enabled by feature `openssh-sftp-client/openssh`
 - Add new feature `openssh` to `openssh-sftp-error`
   and enable it when feature "openssh-sftp-client/openssh" is enabled.
 - Run `cargo-clippy` with `--all-features` in `check.sh`
 - Add new fn `Sftp::from_session`
 - Add integration test for `Sftp::from_session`

Signed-off-by: Jiahao XU <Jiahao_XU@outlook.com>

### CODE REVIEW

**Comment by NobodyXu** [comment]
cc @Xuanwo @silver-ymz This PR adds `Sftp::from_session`.

This function not only makes it easier to create `Sftp` from `openssh::Session`, but also awaits on `openssh::RemoteChild::wait` and `openssh::Session::close` to return more information about any failure, if possible.

Plus, this PR adds integration test with `openssh`.

File: src/sftp/openssh_session.rs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+impl Sftp {
+    /// Create [`Sftp`] from [`openssh::Session`].
+    ///
+    /// Calling [`Sftp::close`] on sftp instances created using this function
+    /// would also await on [`openssh::RemoteChild::wait`] and
+    /// [`openssh::Session::close`] and propagate their error in
+    /// [`Sftp::close`].
+    pub async fn from_session(
```
Comment (Xuanwo) [inline src/sftp/openssh_session.rs:49]: Nice API!

Comment (NobodyXu) [reply src/sftp/openssh_session.rs:49]: Thanks!

**Comment by NobodyXu** [comment]
I would leave this open for a while for feedback.

**Pull request merged**